File: frontend/src/types/job.ts
export interface Job {
  id: number
  jobName: string
  companyName?: string
  companyType?: string
  salaryMin?: number
  salaryMax?: number
  city?: string
  province?: string
  educationReq?: string
  experienceReq?: string
  skillsRequired?: string[]
  description?: string
  requirements?: string
  benefits?: string
  isCivilService?: number
  isStateOwned?: number
  publishDate?: string
  deadline?: string
  categoryName?: string
  viewCount?: number
  hotScore?: number
}

export interface JobSearchParams {
  keyword?: string
  categoryId?: number
  city?: string
  educationReq?: string
  companyType?: string
  page?: number
  size?: number
}

export interface JobCategory {
  id: number
  parentId: number
  name: string
  level: number
  icon?: string
  sortOrder: number
  description?: string
  children?: JobCategory[]
}

File: backend/src/main/java/com/careernavigator/entity/Job.java
package com.careernavigator.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("job")
public class Job {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long categoryId;

    private String jobName;

    private String companyName;

    private String companyType;

    private BigDecimal salaryMin;

    private BigDecimal salaryMax;

    private String city;

    private String province;

    private String educationReq;

    private String experienceReq;

    private String skillsRequired;

    private String description;

    private String requirements;

    private String benefits;

    private Integer isCivilService;

    private Integer isStateOwned;

    private LocalDate publishDate;

    private LocalDate deadline;

    private String source;

    private String sourceUrl;

    private Integer viewCount;

    private Integer hotScore;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private String categoryName;
}
